cmd/mt-control-server/controlmodels/index.go

```diff
@@ -0,0 +1,26 @@
+package controlmodels
+
+type IndexDelByQueryReq struct {
+	Expr      []string `json:"expr" form:"expr" binding:"Required"`
+	OlderThan int64    `json:"olderThan" form:"olderThan"`
+	Method    string   `json:"method" form:"method" binding:";In(dry-run,delete,archive)"`
+	Limit     int      `json:"limit" form:"limit"`
+}
+
+type IndexDelByQueryResp struct {
+	Count      int         `json:"count"`
+	Partitions map[int]int `json:"partitions"`
+}
+
+type IndexRestoreReq struct {
+	Expr             []string `json:"expr" binding:"Required"`
+	From             int64    `json:"from" binding:"Default(0)"`
+	To               int64    `json:"to" binding:"Default(0)"`
+	LastUpdateOffset int64    `json:"lastUpdateOffset" binding:"Default(0)"`
```

Comment:
i wonder if it would happen that defs to be restored have a wide range of lastUpdate values (seems this could easily happen when metrics "expired" at different times but match the same pattern), and just adding an offset may not be ideal for all of them.
in such cases, perhaps an instruction to set LastUpdate to a specific value might be more useful.

Reply:
I originally had that, but I didn't want to add *too* many options upfront.  This is easy enough to iterate on.